solana/gatekeeper-lib/src/service/GatekeeperService.ts

```diff
@@ -88,26 +87,26 @@ export class GatekeeperService {
    * @param {Transaction} transaction
    * @returns {Promise<string>}
    */
-  private async serializeBuiltTransaction(
+  private async signAndSerializeBuiltTransaction(
     transaction: Transaction
   ): Promise<string> {
     // recent blockhash and feepayer are required for serialization
-    // a real recent blockhash will be added in signAndSendTransaction
-    transaction.recentBlockhash = dummyBlockhash;
+    // a real recent blockhash will be added in sendSerializedTransaction
+    const recentBlockhash = await this.connection.getRecentBlockhash();
+    transaction.recentBlockhash = recentBlockhash.blockhash;
     transaction.feePayer = this.payer.publicKey;
-    const unsignedSerializedTx = await transaction
+    await transaction.sign(this.payer, this.gatekeeperAuthority);
+    const serializedTx = await transaction
       .serialize({
         verifySignatures: false,
```

Comment:
Should we now be setting these to true since we have the signature?

Reply:
you're right, updated